ir/src/symbol_table.rs

```diff
@@ -249,6 +280,16 @@ impl SymbolTable {
                     ))
                 }
             }
+            IdentifierType::RandomValue(_offset, size) => {
+                if vector_access.idx() >= *size {
+                    Err(SemanticError::vector_access_out_of_bounds(
+                        vector_access,
+                        *size,
+                    ))
+                } else {
+                    Ok(symbol_type)
+                }
```

Comment:
nit: We should probably keep the order of comparison similar to other match arms just to be consistent, ie,
```
if vector_access.idx() < *size {
    Ok(symbol_type)
} else {
    Err(SemanticError::vector_access_out_of_bounds(
        vector_access,
        *size,
    ))
}
```

or change the order for the others if you have preference.